src/control/server/ctl_ranks_rpc_test.go

```diff
@@ -353,14 +333,9 @@ func TestServer_CtlSvc_StopRanks(t *testing.T) {
 			)
 			svc := mockControlService(t, log, cfg, nil, nil, nil)
 
-			if tc.ctxTimeout == 0 {
-				tc.ctxTimeout = 500 * time.Millisecond
-			}
-			ctx, cancel := context.WithTimeout(context.Background(), tc.ctxTimeout)
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
I think we should not be uniformly setting a 1 second timeout. All of the test cases return deterministically except the timeout test, right? I think this should be conditional -- have a time.Duration in the test params. If timeout == 0, don't set a timeout. Otherwise, set the value on the context. I agree with Mike's suggestion, use a very low timeout value like 1ns.

Reply:
I applied those changes onto one of the tests but it sounds like this approach is going to be canned anyway as per discussion with @mjmac so I'm waiting confirmation before making further changes, I will move the PR state to draft

Reply:
@NiuYawei reminded me that this is needed for 2.2 so I have addressed review comments and re-pushed